File: fizzbuzz/fizzbuzz_2.erl
#!/usr/local/bin/escript
-mode(compile).


main([]) ->
    fizzbuzz(100, 1);
main([MaxStr]) ->
    {Max, _} = string:to_integer(MaxStr),
    fizzbuzz(Max, 1).

fizzbuzz(Max, X) when X > Max ->
    ok;
fizzbuzz(Max, X) when ((X rem 5) == 0) and ((X rem 3) == 0) ->
    io:format("fizzbuzz~n"),
    fizzbuzz(Max, X + 1);
fizzbuzz(Max, X) when ((X rem 3) == 0) ->
    io:format("fizz~n"),
    fizzbuzz(Max, X + 1);
fizzbuzz(Max, X) when ((X rem 5) == 0) ->
    io:format("buzz~n"),
    fizzbuzz(Max, X + 1);
fizzbuzz(Max, X) ->
    io:format("~p~n", [X]),
    fizzbuzz(Max, X + 1).

File: fizzbuzz/fizzbuzz_2.py
#!/usr/bin/python


def fizzbuzz(x):
    if ((x % 5) == 0) and ((x % 3) == 0):
        return "fizzbuzz"
    elif (x % 3) == 0:
        return "fizz"
    elif ((x % 5) == 0):
        return "buzz"
    else:
        return x

max = 100
for x in range(1,max+1):
        print fizzbuzz(x)

